### PR TITLE
fix: Unmark daemoned nodes after stopping them

### PR DESCRIPTION
**Pull request by simster7**
Fixes: https://github.com/argoproj/argo/issues/5002

**Still need testing**

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: test/e2e/daemon_pod_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/argoproj/argo/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/v3/test/e2e/fixtures"
+)
+
+type DaemonPodSuite struct {
+	fixtures.E2ESuite
+}
+
+func (s *DaemonPodSuite) TestWorkflowCompletesIfContainsDaemonPod() {
+	hasSucceeded := func(wf *v1alpha1.Workflow) bool {
```
Comment (alexec) [inline test/e2e/daemon_pod_test.go:21]: Move to fixtures, I’d use a lot!

File: test/e2e/daemon_pod_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    container:
+      image: argoproj/argosay:v2
+      args: ["echo", "hello world"]
+`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.Condition(hasSucceeded), "to show succeeded").
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
+			assert.False(t, status.FinishedAt.IsZero())
```
Comment (alexec) [inline test/e2e/daemon_pod_test.go:57]: Redundant?

File: workflow/controller/exec_control.go [hunk 3]
```diff
@@ -120,6 +120,10 @@ func (woc *wfOperationCtx) killDaemonedChildren(ctx context.Context, nodeID stri
 				firstErr = err
 			}
 		}
+
+		childNode.Daemoned = nil
+		woc.wf.Status.Nodes[childNode.ID] = childNode
```
Comment (alexec) [inline workflow/controller/exec_control.go:125]: Unit test?

**Review by alexec — APPROVED** [review]
conditional approval. See comments.

File: test/e2e/daemon_pod_test.go (hunk 1 above)
Comment (simster7) [reply test/e2e/daemon_pod_test.go:21]: Will do

File: workflow/controller/exec_control.go (hunk 3 above)
Comment (simster7) [reply workflow/controller/exec_control.go:125]: Will do

File: test/e2e/daemon_pod_test.go (hunk 2 above)
Comment (simster7) [reply test/e2e/daemon_pod_test.go:57]: Not redundant... this was the issue in question

File: test/e2e/fixtures/when.go [hunk 4]
```diff
@@ -172,6 +172,7 @@ var ToBeRunning Condition = func(wf *wfv1.Workflow) bool {
 		return node.Phase == wfv1.NodeRunning
 	})
 }
+var ToBeSucceeded Condition = func(wf *wfv1.Workflow) bool {return wf.Status.Phase == wfv1.WorkflowSucceeded }
```
Comment (alexec) [inline test/e2e/fixtures/when.go:175]: nice

**Pull request merged**